### ISSUE TITLE
TypeError: can only concatenate str (not "PosixPath") to str

### ISSUE THREAD

**Issue by tlzhaoboling** [SenJia/EML-NET-Saliency#7]
执行命令python train_resnet.py ~/salicon imagenet_resnet 
<img width="1369" alt="image" src="https://user-images.githubusercontent.com/97796177/209432065-bc0054ef-6cc1-4d9b-8b69-ab19cb9dcf3c.png">
报错，如何解决呢，万分感谢
以及这个预测，只对单张图片进行预测，我修改路径，无法对目录下所有img进行测试，如何解决呢
<img width="1052" alt="image" src="https://user-images.githubusercontent.com/97796177/209432109-6e704d6e-28f4-4c34-af87-be8e38c29c92.png">
万分感谢

**Comment by Cony-Atalas** [comment]
可能需要自己改写代码了，你有试过使用自己训练的encoder和decoder进行预测吗

**Comment by SenJia** [comment]
Thanks for checking the code for me, I was using string for path in the experiment, then later when I released the code,  I used the pathlib to handle all the file paths. I have quickly added a patch in the data loader file, checking if the argument is an instance of pathlib.Path, and convert the input if necessary. I didn't run the code, but the string compatibility issue should be solved. Let me know if you find any other issues.  The code was written to process a single image at a time, but it should be easy to load multiple images and pass them to the model. Simply loading all the file paths under a folder then adding a for loop between lines 70-72 in "eval_combined.py".
